internal/services/eventgrid/eventgrid_namespace_resource.go

```diff
@@ -0,0 +1,601 @@
+	topicSpacesConfig := namespaces.UpdateTopicSpacesConfigurationInfo{}
+	if input == nil {
+		return &topicSpacesConfig
+	}
+
+	topicSpacesConfig = namespaces.UpdateTopicSpacesConfigurationInfo{
+		State: pointer.To(namespaces.TopicSpacesConfigurationStateEnabled),
+		ClientAuthentication: &namespaces.ClientAuthenticationSettings{
+			AlternativeAuthenticationNameSources: expandAlternativeAuthenticationNameSources(input[0].AlternativeAuthenticationNameSources),
+		},
+		MaximumClientSessionsPerAuthenticationName: pointer.To(input[0].MaximumClientSessionsPerAuthenticationName),
+		MaximumSessionExpiryInHours:                pointer.To(input[0].MaximumSessionExpiryInHours),
+		RouteTopicResourceId:                       pointer.To(input[0].RouteTopicResourceId),
+		RoutingEnrichments: &namespaces.RoutingEnrichments{
+			Dynamic: expandDynamicRoutingEnrichments(input[0].DynamicRoutingEnrichment),
+			Static:  expandStaticRoutingEnrichments(input[0].StaticRoutingEnrichment),
+		},
+	}
+
+	return &topicSpacesConfig
+
+}
+
+func expandAlternativeAuthenticationNameSources(input []string) *[]namespaces.AlternativeAuthenticationNameSource {
+	var nameSources []namespaces.AlternativeAuthenticationNameSource
+
+	for _, v := range input {
+		nameSources = append(nameSources, namespaces.AlternativeAuthenticationNameSource(v))
+	}
+	return &nameSources
+}
+
+func expandDynamicRoutingEnrichments(input []RoutingEnrichmentModel) *[]namespaces.DynamicRoutingEnrichment {
+	var dynamicRoutingEnrichments []namespaces.DynamicRoutingEnrichment
+	if len(input) == 0 {
+		return &dynamicRoutingEnrichments
+	}
+
+	for _, v := range input {
+		dynamicRoutingEnrichments = append(dynamicRoutingEnrichments, namespaces.DynamicRoutingEnrichment{
+			Value: pointer.To(v.Value),
+			Key:   pointer.To(v.Key),
+		})
+	}
+
+	return &dynamicRoutingEnrichments
+}
+
+func expandStaticRoutingEnrichments(input []RoutingEnrichmentModel) *[]namespaces.StaticRoutingEnrichment {
+	var staticRoutingEnrichments []namespaces.StaticRoutingEnrichment
+	if len(input) == 0 {
+		return &staticRoutingEnrichments
+	}
+
+	for _, v := range input {
+		staticRoutingEnrichments = append(staticRoutingEnrichments, namespaces.StaticStringRoutingEnrichment{
+			Value:     pointer.To(v.Value),
+			Key:       pointer.To(v.Key),
+			ValueType: namespaces.StaticRoutingEnrichmentTypeString,
+		})
+	}
+
+	return &staticRoutingEnrichments
+}
+
+func flattenTopicSpacesConfiguration(topicSpacesConfig *namespaces.TopicSpacesConfiguration) []TopicSpacesConfigurationModel {
+	var output TopicSpacesConfigurationModel
+	if topicSpacesConfig == nil {
+		return nil
+	}
+
+	output.MaximumSessionExpiryInHours = pointer.From(topicSpacesConfig.MaximumSessionExpiryInHours)
+	output.MaximumClientSessionsPerAuthenticationName = pointer.From(topicSpacesConfig.MaximumClientSessionsPerAuthenticationName)
+	output.RouteTopicResourceId = pointer.From(topicSpacesConfig.RouteTopicResourceId)
```

Comment:
We should parse this before setting into state so that we can catch any potential casing inconsistencies in the API.